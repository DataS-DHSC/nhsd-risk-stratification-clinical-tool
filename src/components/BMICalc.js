import React from 'react';
import PropTypes from 'prop-types';
import CalcBMI from '../utils/calcBMI';

const BMICalc = ({ field, linkedValues, groupName }) => {
  const arg1 = linkedValues.find((item) => item.groupName === field.links[0]);
  const arg2 = linkedValues.find((item) => item.groupName === field.links[1]);

  const calculatedBmi = CalcBMI(arg1.value, arg2.value, 2);
  /* Ensure the continue button doesn't bounce around when BMI is computed. */
  const legend = calculatedBmi ? field.label : '\u00A0';
  return (
    <p name={groupName} id={groupName}>
      {legend}
      <strong>{arg1 && arg2 ? calculatedBmi : ''}</strong>
    </p>
  );
};

BMICalc.propTypes = {
  field: PropTypes.shape().isRequired,
  linkedValues: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  groupName: PropTypes.string.isRequired,
};

export default BMICalc;
