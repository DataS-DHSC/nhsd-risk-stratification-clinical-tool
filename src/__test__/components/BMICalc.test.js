import React from 'react';
import { render } from '@testing-library/react';
import BMICalc from '../../components/BMICalc';

const setup = () => {
  const props = {
    field: {
      label: "The patient's BMI score is:",
      links: ['weight', 'height'],
    },
    prevData: undefined,
    linkedValues: [
      { groupName: 'height', value: 140 },
      { groupName: 'weight', value: 77 },
    ],
    groupName: 'bmi-calc',
  };

  return render(<BMICalc {...props} />);
};

describe('BMICalc', () => {
  it('should match snapshot', () => {
    const container = setup();
    expect(container.asFragment()).toMatchSnapshot();
  });
});
