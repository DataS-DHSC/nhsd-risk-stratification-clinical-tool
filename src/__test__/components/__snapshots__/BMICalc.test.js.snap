// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BMICalc should match snapshot 1`] = `
<DocumentFragment>
  <p
    id="bmi-calc"
    name="bmi-calc"
  >
    The patient's BMI score is:
    <strong>
      39.29
    </strong>
  </p>
</DocumentFragment>
`;
